NuRadioMC/SignalProp/Simple_radiopropa_tracer.py

```diff
@@ -0,0 +1,978 @@
+            spec[1] *= reflection_coefficient * np.exp(1j * phase_shift)
+            spec[2] *= reflection_coefficient * np.exp(1j * phase_shift)
+            self.__logger.debug(
+                f"ray is reflecting {i_reflections:d} times at the bottom -> reducing the signal by a factor of {reflection_coefficient:.2f}")
+
+
+        ## apply focussing effect
+        if self.__config['propagation']['focusing']:
+            focusing = self.get_focusing(i_solution, limit=float(self.__config['propagation']['focusing_limit']))
+            spec[1:] *= focusing
+
+        efield.set_frequency_spectrum(spec, efield.get_sampling_rate())
+        return efield
+
+
+    def get_output_parameters(self):
+        """
+        Returns a list with information about parameters to include in the output data structure that are specific
+        to this raytracer
+
+        ! be sure that the first entry is specific to your raytracer !
+
+        Returns:
+        -----------------
+        list with entries of form [{'name': str, 'ndim': int}]
+            ! be sure that the first entry is specific to your raytracer !
+            'name': Name of the new parameter to include in the data structure
+            'ndim': Dimension of the data structure for the parameter
+        """
+        return [
+            {'name': 'sphere_sizes','ndim':len(self.__sphere_sizes)},
+            {'name': 'launch_vector', 'ndim': 3},
+            {'name': 'focusing_factor', 'ndim': 1},
+            {'name': 'ray_tracing_reflection', 'ndim': 1},
+            {'name': 'ray_tracing_reflection_case', 'ndim': 1},
+            {'name': 'ray_tracing_solution_type', 'ndim': 1}
+        ]
+
+    def get_raytracing_output(self,i_solution):
+        """
+        Write parameters that are specific to this raytracer into the output data.
+
+        Parameters:
+        ---------------
+        i_solution: int
+            The index of the raytracing solution
+
+        Returns:
+        ---------------
+        dictionary with the keys matching the parameter names specified in get_output_parameters and the values being
+        the results from the raytracing
+        """
+        if self.__config['propagation']['focusing']:    
+            focusing = self.get_focusing(i_solution, limit=float(self.__config['propagation']['focusing_limit']))
+        else: 
+            focusing = 1
+        output_dict = {
+            'sphere_sizes': self.__sphere_sizes,
+            'launch_vector': self.get_launch_vector(i_solution),
+            'focusing_factor': focusing,
+            'ray_tracing_reflection': self.get_results()[i_solution]['reflection'],
+            'ray_tracing_reflection_case': self.get_results()[i_solution]['reflection_case'],
+            'ray_tracing_solution_type': self.get_solution_type(i_solution)            
+        }
+        return output_dict
+
+
+    def get_number_of_raytracing_solutions(self):
```

Comment:
This isn't right for radiopropa, is it?

Reply:
You don't expect more for now, do you? If so, then this property is dependant of the ice model and not the tracer and should this maybe be moved?

Reply:
I thought the whole point of RadioPropa was that you could use more complicated ice models?
I don't really know what the best way to solve this is. It determines the size of the output arrays, so you would need to know ahead of time how many solutions you can have. But you don't necessarily know that.
Maybe have the user configure this by hand and if you find too many solutions select the ones with the largest E-field amplitude?

Reply:
I'll open an issue for this and then we can discuss it on Thursday